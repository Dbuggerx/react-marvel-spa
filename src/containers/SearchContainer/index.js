// @flow

import React, { Component } from 'react';
import SearchForm from '../../components/SearchForm';

type Props = {
    onSearchChanged: (param: string) => void,
    lastSearch: string,
};

/**
 * Handles search data
 */
export default class SearchContainer extends Component<Props> {
    handleSearch = (param: string) => {
        if (param.toLowerCase() !== this.props.lastSearch) this.props.onSearchChanged(param);
    };

    render() {
        return <SearchForm lastSearch={this.props.lastSearch} onSearch={this.handleSearch} />;
    }
}
