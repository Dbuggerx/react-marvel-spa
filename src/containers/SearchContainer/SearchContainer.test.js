import React from 'react';
import { shallow } from 'enzyme';
import SearchContainer from './index';

describe('SearchContainer', () => {
    describe('render', () => {
        test('renders properly', () => {
            const wrapper = shallow(<SearchContainer onSearchChanged={jest.fn()} />);
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('handleSearch', () => {
        test('it calls "onSearchChanged"', () => {
            const handleSearchChangedMock = jest.fn();
            const wrapper = shallow(<SearchContainer onSearchChanged={handleSearchChangedMock} />);
            wrapper.instance().handleSearch('test');
            expect(handleSearchChangedMock).toHaveBeenCalledTimes(1);
            expect(handleSearchChangedMock).toHaveBeenCalledWith('test');
        });

        test('it does not call "onSearchChanged"', () => {
            const handleSearchChangedMock = jest.fn();
            const wrapper = shallow(<SearchContainer onSearchChanged={handleSearchChangedMock} />);
            wrapper.instance().lastSearch = 'test';
            wrapper.instance().handleSearch('test');
            expect(handleSearchChangedMock).not.toHaveBeenCalled();
        });
    });
});
