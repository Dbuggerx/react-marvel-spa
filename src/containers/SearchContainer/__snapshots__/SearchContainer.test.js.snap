// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchContainer render renders properly 1`] = `
<SearchForm
  onSearch={[Function]}
/>
`;
